draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -239,66 +233,59 @@
+      </t>
+      <t>
+        A compliant AS, including a Route Server AS (RS AS), MUST have an ASPA.
+        An RS AS SHOULD register an AS 0 ASPA.
+        An AS SHOULD NOT have more than one ASPA.
+        Such a practice helps prevent race conditions during ASPA updates that might affect prefix propagation.
+        The software that provides hosting for ASPA records SHOULD support enforcement of this practice.
+        During a transition process between different certificate authority (CA) registries, the ASPA records SHOULD be kept identical in all relevant registries.
+      </t>
+      <t>
+        As mentioned in <xref target="ASPA"/>, the set of provider ASes contained in the VAP(s) is referred to as the VAP-SPAS of the AS registering the ASPA(s).
+        Normally, a VAP-SPAS is not expected to contain both an AS 0 and other Provider ASes,
+        but an unexpected presence of AS 0 has no influence on the AS path verification procedures (see <xref target="pair-validation"/>, <xref target="verif"/>).
+      </t>
+      <t>
+        Each of the two ASes in a mutual-transit pair MUST register its ASPA including the other AS in its SPAS.
+      </t>
+      <t>
+        The ASes on the boundary of an AS Confederation MUST register ASPAs using the Confederation's global ASN as the CAS.
+      </t>
+    </section>
+    <section title="Provider Authorization Function" anchor="pair-validation">
+      <t>
+          Let AS x and AS y represent two unique ASes.
+          A provider authorization function, auth(AS x, AS y), checks if the ordered pair of ASNs, (AS x, AS y), has the property that AS y is an attested provider of AS x per VAP-SPAS of AS x.
+          The VAP-SPAS table is assumed to be organized in such a way that it can be queried to check (1) if a specified CAS = AS x has an entry (i.e., SPAS listed), or (2) if for a given (AS x, AS y) tuple, AS x is listed in the VAP-SPAS as a provider associated with CAS = AS y.
+          By term "Provider+" the function should signal if AS y plays role of Provider, non-transparent RS, or mutual-transit neighbor.
           This function is specified as follows:
       </t>
       <t>
         <figure anchor="fig1" align="left" suppress-title="true" pn="figure-1">
-          <name slugifiedName="HFC-illustration">Hop-check function.</name>
+          <name slugifiedName="HFC-illustration">provider authorization function.</name>
           <artwork align="left" name="" type="" alt="">
 <![CDATA[
 
                      /
                      | "No Attestation" if there is no entry
-                     |   in VAP-SPAS table for CAS = AS(i)
+                     |   in VAP-SPAS table for CAS = AS x
                      |
-hop(AS(i), AS(j)) =  / Else, "Provider+" if the VAP-SPAS entry
-                     \   for CAS = AS(i) includes AS(j)
+auth(AS x, AS y) =  / Else, "Provider+" if the VAP-SPAS entry
```

Comment:
I think `auth` is misleading. auth = authorized, authenticated ?

Suggestions: `relation(AS x, AS y)`, `query(AS x, AS y)` or just stick with hop - naming it hop was totally fine in my mind.

Reply:
hop has its own meaning, that it is hop. :) I would prefer authorized than, without short words.

Reply:
After consulting with @waehlisch, we would like to suggest the following  change:
```
                                             /
                                            | "No Attestation" if there is no valid ASPA object for customer x,
                                            |   i.e., no entry in VAP-SPAS table for CAS = AS x.
                                            |
   authorized(AS x, AS y) =  / Else, "Provider+" if provider y is included in a valid ASPA object of customer x,
                                            \   i.e., there is the VAP-SPAS entry for CAS = AS x includes AS y.
                                            |
                                            | Else, "Not Provider+"
                                            \
```

1.) changing signature of the function to: `authorized(AS customer, AS provider)` 

2.)

Reply:
```
   Let AS x and AS y represent two unique ASes.  The provider
   authorization function, authorized(AS x, AS y), checks if  ASNs AS x, AS y, 
   has the property that AS y is an attested
   provider of AS x per VAP-SPAS of AS x. This function is
   specified as follows:
   
   <function>
   
   By term "Provider+" the
   function should signal if AS y plays role of Provider, non-
   transparent RS, or mutual-transit neighbor. 
   
   ```

Reply:
We think the section had a lot of redundant content, so we decided to shrink it down.

Reply:
We have the notation AS x, AS y in the definition of ASPA:

(AS x, {AS y1, AS y2, ...})

I am not strictly against changing x -> cutsomer, y -> provider, but I am not sure if it is improving readability since it is not strictly speaking a provider, it's provider+ (provider or RS).